bootstrap.sh

```diff
@@ -120,165 +307,111 @@ echo "Enter your domain name"
+root_host=
+until [[ ${root_host} =~ ^[a-z0-9\.-]*$ && -n ${root_host} ]]; do
+	[[ -n ${root_host} ]] && echo "Invalid domain name"
+	read -r -p "Domain name: " root_host
 done
 
-public_ip=$(curl -s ipinfo.io/ip)
-domain_ip=$(dig +short @1.1.1.1 ${root_host})
-
-until [[ $domain_ip =~ $public_ip ]]; do
-  echo
-  echo "The domain $root_host does not resolve to the public IP of this server ($public_ip)"
-  echo
-  root_host_prev=$root_host
-  read -p "Domain name [$root_host_prev]: " root_host
-  if [ -z ${root_host} ]; then
-    root_host=$root_host_prev
-  fi
-  public_ip=$(curl -s ipinfo.io/ip)
-  domain_ip=$(dig +short @1.1.1.1 ${root_host})
-  echo
-done
 
 echo
-echo "Running certbot in dry-run mode to test the validity of the domain..."
-$SUDO certbot certonly --non-interactive --break-my-certs --force-renewal --agree-tos --email root@localhost.com --standalone --staging -d $root_host -d wg.$root_host -d auth.$root_host || exit
-echo "OK"
-
-echo "root_host: \"${root_host}\"" >> $HOME/ansible-easy-vpn/custom.yml
-
-
-if [[ ! $aws =~ true ]]; then
-  echo
-  echo "Would you like to use an existing SSH key?"
-  echo "Press 'n' if you want to generate a new SSH key pair"
-  echo
-  read -p "Use existing SSH key? [y/N]: " new_ssh_key_pair
-  until [[ "$new_ssh_key_pair" =~ ^[yYnN]*$ ]]; do
-          echo "$new_ssh_key_pair: invalid selection."
-          read -p "[y/N]: " new_ssh_key_pair
-  done
-  echo "enable_ssh_keygen: true" >> $HOME/ansible-easy-vpn/custom.yml
-
-  if [[ "$new_ssh_key_pair" =~ ^[yY]$ ]]; then
-    echo
-    read -p "Please enter your SSH public key: " ssh_key_pair
-
-    echo "ssh_public_key: \"${ssh_key_pair}\"" >> $HOME/ansible-easy-vpn/custom.yml
-  fi
-else
-  echo
-  read -p "Are you running this script on an AWS EC2 instance? [y/N]: " aws_ec2
-  until [[ "$aws_ec2" =~ ^[yYnN]*$ ]]; do
-          echo "$aws_ec2: invalid selection."
-          read -p "[y/N]: " aws_ec2
-  done
-  if [[ "$aws_ec2" =~ ^[yY]$ ]]; then
-    export AWS_EC2=true
-  echo
-  echo "Please use the SSH keys that you specified in the AWS Management Console to log in to the server."
-  echo "Also, make sure that your Security Group allows inbound connections on 51820/udp, 80/tcp and 443/tcp."
-  echo
-  fi
-fi
+[[ -f /etc/resolv.conf ]] && {
+	echo "Entries in /etc/resolv.conf for nameserver(s)"
+	grep ^nameserver /etc/resolv.conf
+	echo
+}
+read -r -p "Enter the IP address of the Nameserver to use for DNS queires [1.1.1.1]: " DNS_HOST
+DNS_HOST=${DNS_HOST:-1.1.1.1}
+#declare -p DNS_HOST
+echo
+
+while :
+do
+	# Okay, a "list of IPs" probably doesn't make sense for WG,
+	# but checking for the sub domains does make sense.
+	# - got this using main domain for sub domains right now
+	# Probably can't do round robin DNS for WG ....
+	public_ip=$(curl -s ipinfo.io/ip)
+	domain_ip_list=$(get_ip_list "${root_host}")
+	wg_domain_ip_list=$(get_ip_list "${root_host}")
+	auth_domain_ip_list=$(get_ip_list "${root_host}")
```

Comment:
I don't understand this part. I think you also wanted to check wg.domain.com and auth.domain.com, but later commented these out?
In the current implementation, doesn't it just accomplish the same thing as the original script?

Reply:
Yes, more work for another time.  The whole round robin thing is probably no value for WG's case, but I can see this being extended for other "types" of server installs.  This is, after all, just the "bootstrap" to get things going before the ansible works being late in the script.

Reply:
Added sub domains back in.